apps/meteor/app/livechat/imports/server/rest/dashboards.ts

```diff
@@ -195,13 +203,17 @@ API.v1.addRoute(
 			if (isNaN(Date.parse(start))) {
 				return API.v1.failure('The "start" query parameter must be a valid date.');
 			}
-			start = new Date(start);
+			const startDate = new Date(start);
 
 			if (isNaN(Date.parse(end))) {
 				return API.v1.failure('The "end" query parameter must be a valid date.');
 			}
-			end = new Date(end);
-			const result = findAllChatMetricsByAgent({ start, end, departmentId });
+			const endDate = new Date(end);
+
+			const result = findAllChatMetricsByAgent({ start: startDate, end: endDate, departmentId }) as OperationResult<
```

Comment:
Don't feel quite comfortable with this casting 😬 why do you need the "OperationResult", if it should be based on the type you're about to return

Reply:
Alright, I declared a new type within this file for the response to avoid using `OperationResult`.